engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java

```diff
@@ -150,7 +152,13 @@ private String asFeelContextExpression(
     return new MappingContextVisitor<>() {
       @Override
       public String onEntry(final String targetKey, final Expression sourceExpression) {
-        return targetKey + ":" + sourceExpression.getExpression();
+        String expression = sourceExpression.getExpression();
+
+        if (sourceExpression instanceof StaticExpression) {
+          expression = String.format("\"%s\"", StringEscapeUtils.escapeJava(expression));
```

Comment:
🔧 the feel syntax is closer to JSON than it is to Java. I recommend escaping it for JSON usage.

```suggestion
          expression = String.format("\"%s\"", StringEscapeUtils.escapeJson(expression));
```

Reply:
🔧 That makes sense, please add this reasoning to the commit message

Reply:
I added it to the explanation in the commit message and this PR